atomic_reactor/util.py

```diff
@@ -714,17 +715,24 @@ def get_manifest_digests(image, registry, insecure=False, dockercfg_path=None,
     :param insecure: bool, when True registry's cert is not verified
     :param dockercfg_path: str, dirname of .dockercfg location
     :param versions: tuple, which manifest schema versions to fetch digest
+    :param throw_error: bool, when True exception is thrown if no digests found
 
     :return: dict, versions mapped to their digest
     """
     digests = {}
     for version in versions:
         media_type = get_manifest_media_type(version)
         headers = {'Accept': media_type}
-        response = query_registry(
-            image, registry, digest=None,
-            insecure=insecure, dockercfg_path=dockercfg_path,
-            version=version)
+
+        for attempts in range(10):
+            response = query_registry(
+                image, registry, digest=None,
+                insecure=insecure, dockercfg_path=dockercfg_path,
+                version=version)
+            if response.status_code == 404:
+                sleep(1)
+            else:
+                break
```

Comment:
In some cases the docker/distribution code already returns 404 if it can't satisfy the Accept header (manifest list where an appropriate image can't be extracted, for example), and the current draft of OCI support extends that behavior. This looks like a good recipe for slowing builds down by 10 or 20 seconds.

Is this some sort of race condition workaround? A workaround for a bug in some component? In general, there should be never any reason to retry a 404 - it's not a '503 Service Unavailable' or something.